WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -1209,22 +1222,40 @@ private extension EditableOrderViewModel {
             .assign(to: &$multipleLinesMessage)
     }
 
-    func retrieveTaxBasedOnSetting() {
+    func configureTaxRates() {
         stores.dispatch(SettingAction.retrieveTaxBasedOnSetting(siteID: siteID,
                                                                 onCompletion: { [weak self] result in
             guard let self = self else { return }
 
             switch result {
                 case .success(let setting):
-                self.shouldShowNewTaxRateSection = self.featureFlagService.isFeatureFlagEnabled(.manualTaxesInOrderM2) &&
-                (setting == .customerBillingAddress || setting == .customerShippingAddress)
                 self.taxBasedOnSetting = setting
+
+                let canApplyTaxRates = self.featureFlagService.isFeatureFlagEnabled(.manualTaxesInOrderM2) &&
+                (setting == .customerBillingAddress || setting == .customerShippingAddress)
+                if canApplyTaxRates {
+                    Task { @MainActor in
+                        await self.applySelectedStoredTaxRateIfAny()
+                        // Show the tax rate section once we know if any stored tax rate applies, as it can change the text
+                        self.shouldShowNewTaxRateSection = true
+                    }
+                }
+
                 case .failure(let error):
                 DDLogError("⛔️ Error retrieving tax based on setting: \(error)")
             }
         }))
     }
 
+    func applySelectedStoredTaxRateIfAny() async {
```

Comment:
We can make this one private:
```suggestion
    private func applySelectedStoredTaxRateIfAny() async {
```

Reply:
It's already part of a private extension 🕺

Reply:
Oops, I didn't see it! 😅